src/Microsoft.AspNetCore.Server.IntegrationTesting/Deployers/IISDeployer.cs

```diff
@@ -36,13 +36,15 @@ public override DeploymentResult Deploy()
 
             _application = new IISApplication(DeploymentParameters, Logger);
 
-            // Publish to IIS root\application folder.
-            DotnetPublish(publishRoot: _application.WebSiteRootFolder);
+            if (DeploymentParameters.PublishApplicationBeforeDeployment)
```

Comment:
DeploymentParameters.PublishApplicationBeforeDeployment is set to true at the line 35. 
So, adding this if condition would not affect any change. IISExpress has the same code logic.